extending/directive/implementation.rst

```diff
@@ -0,0 +1,51 @@
+Implementing a Directive
+------------------------
+
+To create your own a directive subclass `gyro.core.directive.DirectiveProcessor <https://github.com/perfectsense/gyro/blob/master/core/src/main/java/gyro/core/directive/DirectiveProcessor.java>`_ and implement the method **process**.
```

Comment:
Need to explain how the generic affects a directive usage. `DirectiveProcessor<S extends Scope>` takes a `Scope` as a generic. This limits the scope that a generic can be used in:

`Scope` - Directive can be used anywhere but it's still the directives responsibility to see what scope it was given and adjust accordingly. For example, there may be cases where you have to extend using `Scope` so that your directive will work inside a Resource or inside a file, but it doesn't work in `.gyro/init.gyro`.
`DiffableScope` - Directive can only be used inside a Resource definition.
`RootScope` - Directive can only be used inside `.gyro/init.gyro`.
`FileScope` - Directive can only be used inside a Gyro file (excluding `.gyro/init.gyro`) but cannot be used inside a Resource within that file.